modules/ebs_snapshot_explorer/main.py

```diff
@@ -0,0 +1,303 @@
+    return failed_volumes
+
+
+def delete_snapshots(client, snapshots):
+    """Deletes a given list of snapshots"""
+    failed_snapshots = []
+    for snapshot in snapshots:
+        try:
+            client.delete_snapshot(SnapshotId=snapshot)
+        except ClientError:
+            failed_snapshots.append(snapshot)
+    return failed_snapshots
+
+
+def cleanup(client):
+    """Cleans up the temporary snapshots and volumes created during this
+    modules execution
+    """
+    new_data = {}
+    success = True
+    temp_file = Path(__file__).parent / 'temp.json'
+    if temp_file.is_file():
+        with temp_file.open('r') as file:
+            data = json.load(file)
+            if 'snapshots' in data:
+                new_data['snapshots'] = delete_snapshots(client, data['snapshots'])
+            if 'volumes' in data:
+                new_data['volumes'] = delete_volumes(client, data['volumes'])
+        if 'volumes' in new_data and new_data['volumes']:
+            print('  Failed to delete volumes: {}'.format(new_data['volumes']))
+            success = False
+        if 'snapshots' in new_data and new_data['snapshots']:
+            print('  Failed to delete snapshots: {}'.format(new_data['snapshots']))
+            success = False
+        store_temp_data(new_data)
+        if success:
+            temp_file.unlink()
+    return success
+
+
+def store_temp_data(data):
+    """Stores temporary data in a JSON file"""
+    temp_file = Path(__file__).parent / 'temp.json'
+    if temp_file.exists():
+        with temp_file.open('r') as json_file:
+            existing_data = json.load(json_file)
+            data.update(existing_data)
+    with temp_file.open('w+') as json_file:
+        json.dump(data, json_file)
+
+
+def main(args, pacu):
+    """Main module function, called from Pacu"""
+    args = parser.parse_args(args)
+    session = pacu.get_active_session()
+    print = pacu.print
+
+    instance = args.instance
+    region = args.region
+    zone = region + args.zone
+    client = pacu.get_boto3_client('ec2', region)
+
+    if not cleanup(client):
+        print('  Cleanup failed')
+        return summary_data
+
+    snapshots = get_snapshots(pacu, session, region)
+    volumes = get_volumes(pacu, session, region)
+    summary_data = {'snapshots': len(snapshots), 'volumes': len(volumes)}
+
+    print('  Attaching volumes...')
+    temp_snaps = generate_snapshots_from_volumes(client, volumes)
+    temp_volumes = generate_volumes_from_snapshots(client, temp_snaps, zone)
+    load_volumes(client, pacu.print, pacu.input, instance, temp_volumes)
+    print('  Finished attaching volumes')
+
+    print('  Attaching volumes from existing snapshots')
+    temp_volumes = generate_volumes_from_snapshots(client, snapshots, zone)
+    load_volumes(client, pacu.print, pacu.input, instance, temp_volumes)
+    print('  Finished attaching existing snapshot volumes ')
+
+    return summary_data
+
+
+def summary(data, pacu):
```

Comment:
Looks like `pacu` doesn't need to get passed in here

Reply:
These are the required arguments for the summary function. Could set the default value in Pacu to None so that it's optional.